domain/importtracker.go

```diff
@@ -158,3 +177,40 @@ func (ir *ImportTracker) Deref(ref Reference) (resolvedRef Reference, allowPrivi
 	}
 	return ref, false, false, nil
 }
+
+func verifyLocalPath(path string, pathResultF pathResultFunc) error {
+	earthlyFileName := "Earthfile"
+	res := pathResultF(path)
+	if res == notExist {
+		return hint.Wrapf(errors.Errorf("path %q does not exist", path), "Verify the path %q exists", path)
+	}
+	if res == file {
+		if filepath.Base(path) == earthlyFileName {
+			return hint.Wrapf(errors.Errorf("path %q is not a directory", path), "Did you mean to import %q?", filepath.Dir(path))
+		}
+		return hint.Wrap(errors.Errorf("path %q is not a directory", path), "Please use a directory when using a local IMPORT path")
+	}
+	res = pathResultF(filepath.Join(path, earthlyFileName))
+	if res == notExist {
+		if filepath.Base(path) == earthlyFileName {
+			return hint.Wrapf(errors.Errorf("path %q does not contain an %s", path, earthlyFileName), "The path %q ends with an %q which is a directory.\nDid you mean to create an %q as a file instead?", path, earthlyFileName, earthlyFileName)
+		}
+		return hint.Wrapf(errors.Errorf("path %q does not contain an %s", path, earthlyFileName), "Verify the path %q contains an %s", path, earthlyFileName)
+	}
+	if res == dir {
+		return hint.Wrapf(errors.Errorf("path %q does contains an %s which is not a file", path, earthlyFileName), "The local IMPORT path %q contains an %q directory and not a file", path, earthlyFileName)
+	}
+	return nil
+}
```

Comment:
Does this work when this is in a remote context? E.g. I call the target `github.com/foo/bar+target` and that Earthfile imports `./buz`.

Reply:
Good question.
So it seems to not have a negative or positive impact.
Some options I tested inside the remote Earthfile:
* If the local IMPORT exists - it works.
* If the local IMPORT is an absolute path - it returns an errors (~ absolute paths are not supported in the context of remote ref)
* if the local import is a relative path that does not exist - returns an error (but not as pretty as this PR tries to make it, because it's checked when the path is resolved)

Reply:
Interesting... it might be nice to throw an error for that case too. But that could be a separate PR.